exercises/atbash-cipher/example.rs

```diff
@@ -0,0 +1,36 @@
+use std::ascii::AsciiExt;
+
+fn ascii(ch: char) -> u8 {
+    ch as u8
+}
+
+fn get_transpose(ch: char) -> char {
+    if ch.is_digit(10) {
+        ch
+    } else {
+        (ascii('z') - ascii(ch) + ascii('a')) as char
+    }
+}
+
+pub fn encode(plaintext: &str) -> String {
+    plaintext
+    .to_lowercase()
+    .chars()
+    .filter(|&ch| ch.is_ascii())
+    .filter(|&ch| ch.is_alphanumeric())
+    .map(|ch| get_transpose(ch))
+    .collect::<Vec<char>>()
+    .chunks(5)
+    .map(|slice| slice.iter().cloned().collect::<String>())
+    .collect::<Vec<String>>()
+    .join(" ")
+}
+
+pub fn decode(ciphertext: &str) -> String {
+    ciphertext
+    .chars()
+    .filter(|&ch| !ch.is_whitespace())
+    .fold("".to_string(), |decoded_result, ch| {
+            decoded_result + &(get_transpose(ch).to_string())
```

Comment:
given that the interior of this `fold` has been simplified greatly since the first iteration of this PR, does it now make sense to make this a `map` then `collect<String>` instead of a `fold`?